### PR TITLE
Refactor XPU layers into python

### PR DESCRIPTION
**Pull request by sramakintel**
## Description

This PR refactors xpu layers-driver and oneapi into a separate base layer in python and is reused by pytorch,jax and tf frameworks
## Related Issue

https://jira.devtools.intel.com/browse/MLOPS-2128
## Changes Made

- 
- [x] The code follows the project's [coding standards](https://github.com/intel/ai-containers/blob/main/CONTRIBUTING.md#code-style).
- [x] No Intel Internal IP is present within the changes.
- [x] The documentation has been updated to reflect any changes in functionality.

## Validation

- [x] I have tested any changes in container groups locally with [`test_runner.py`](https://github.com/intel/ai-containers/blob/main/test-runner/README.md) with all existing tests passing, and I have added new tests where applicable.

### CODE REVIEW

File: python/README.md [hunk 1]
```diff
@@ -10,6 +12,7 @@ The images below include variations for only the core packages in the [Intel® D
 | ---------------------- | ---------- |
 | `3.10-full`, `latest`  | `2024.2.0` |
 | `3.10-core`            | `2024.2.0` |
+| `3.10-xpu-idp`         | `2024.2.1` |
```
Comment (tylertitsworth) [inline python/README.md:15]: Maybe just `3.10-xpu`? Do we have both full and core options for the package? (see overrides file)

File: python/tests/tests.yaml [hunk 2]
```diff
@@ -27,6 +27,13 @@ perf-stock:
   volumes:
   - dst: /tests
     src: $PWD/python/tests
+xpu-base-layers-${PACKAGE_OPTION:-pip}:
```
Comment (tylertitsworth) [inline python/tests/tests.yaml:30]: PACKAGE_OPTION is not a var, it should be IDP_VERSION

**Review by tylertitsworth — CHANGES_REQUESTED** [review]
Great work @sramakintel, I have a small nit regarding the parameterization of the tests. Can you please verify both `full` and `core` python package versions?

File: python/README.md (hunk 1 above)
Comment (sramakintel) [reply python/README.md]: support for `core` and `full` added

File: python/tests/tests.yaml (hunk 2 above)
Comment (sramakintel) [reply python/tests/tests.yaml]: made changes.

File: python/compose.override.yaml [hunk 3]
```diff
@@ -24,3 +24,6 @@ services:
         org.opencontainers.image.title: "Intel® Distribution for Python Base Image"
         org.opencontainers.image.version: 3.10-${IDP_VERSION:-core}
     image: ${REGISTRY}/${REPO}:b-${GITHUB_RUN_NUMBER:-0}-${BASE_IMAGE_NAME:-ubuntu}-${BASE_IMAGE_TAG:-22.04}-idp-${IDP_VERSION:-core}-py${PYTHON_VERSION:-3.10}-base
+  xpu:
```
[the remark this answers is not in the document]
Comment (sramakintel) [reply python/compose.override.yaml:27]: updated the labels

**Pull request merged**